README.md

```diff
@@ -1,145 +1,132 @@
+
+1) Proxy server for the modules defined in vgo
+2) Package registry (like npm and ruby gems) for modules
+
+Athens is a proxy server for [vgo modules](https://github.com/golang/go/wiki/vgo). It implements the download protocol specified [here](https://research.swtch.com/vgo-module) (under "Download Protocol"), and a few additional API endpoints to make it more useful. See [API.md](./API.md) for more information.
+
+# Architecture
+
+Athens is composed roughly of three logical pieces. The below list contains links
+to a description of each:
+
+* [Module proxy](./PROXY.md)
+* [Module registry](./REGISTRY.md)
+* [CLI](./CLI.md)
+
+# Development
+
+The server is written using [Buffalo](https://gobuffalo.io/), so it's fairly straightforward to get started on development. You'll need Buffalo v0.11.0 or later to do devlopment on Athens.
+
+Download
+[v0.11.0](https://github.com/gobuffalo/buffalo/releases/tag/v0.11.0) or later, untar/unzip the binary into your PATH, and then run the following from the root of this repository:
+
+```console
+buffalo dev
+```
+
+You'll see some output in your console that looks like this:
+
+```console
+$ buffalo dev
+buffalo: 2018/02/25 16:09:36 === Rebuild on: :start: ===
+buffalo: 2018/02/25 16:09:36 === Running: go build -v -i -o tmp/vgoprox-build  (PID: 94067) ===
+buffalo: 2018/02/25 16:09:37 === Building Completed (PID: 94067) (Time: 1.115613079s) ===
+buffalo: 2018/02/25 16:09:37 === Running: tmp/vgoprox-build (PID: 94078) ===
+time="2018-02-25T16:09:37-08:00" level=info msg="Starting application at 127.0.0.1:3000"
+INFO[2018-02-25T16:09:37-08:00] Starting Simple Background Worker
+
+Webpack is watching the files…
+```
+
+After the `Starting application at 127.0.0.1:3000` is logged, the server is up and running.
+As you edit and save code, Buffalo will automatically restart the server. **This means that all of your modules will disappear** because the only storage driver is in-memory right now.
+
+See [CLI](#CLI) for information on how to add modules back into the server.
+
+## Dependencies
+
+To run the development server, or run tests (tip: run `make test` to easily run tests), you'll need a running MongoDB server. We plan to add more service dependencies in the future, so we are using
+[Docker](https://www.docker.com/) and
+[Docker Compose](https://docs.docker.com/compose/) to create and destroy development environments.
+
+To create, run the following from the repository root:
+
+```console
+docker-compose up -d
+```
+
+To destroy:
+
+```console
+docker-compose down
+```
+
+# Contributing
+
+This project is early and there's plenty of interesting and challenging work to do.
+
+If you find a bug or want to fix a bug, I :heart: PRs and issues!
+
+Below is a list of general areas that I'm planning to work on, so if you'd like to help there ping me or file an issue (I am 'arschles' on the Gophers Slack):
+
+* New storage backends (probably disk next because it's the easiest on the TODO list :smile:)
```

Comment:
we have issues for mongo (in progress), postrgresql and mysql

Reply:
also, since we now have disk storage implemented, we can remove the part in parentheses. could replace it with some of the issues mentioned above (#40, #41, etc...)